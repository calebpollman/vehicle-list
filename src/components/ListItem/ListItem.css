.link-container {
  text-decoration: none;
  width: 100%;
}

.list-item-container {
  border-bottom: 1px solid var(--color-list-item-background);
  padding: 2%;
  transition: all linear 0.2s;
}

.list-item-container:hover {
  background-color: var(--color-list-item-background);
}

.list-item-title {
  color: var(--color-list-item-text);
  font-family: var(--font-application);
  font-size: var(--font-size-small);
  font-weight: var(--font-weight-bold);
}

.list-item-text {
  color: var(--color-list-item-sub-text);
  font-family: var(--font-application);
  font-size: var(--font-size-small);
}