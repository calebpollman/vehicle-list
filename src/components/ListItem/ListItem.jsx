import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

const ListItem = ({vehicle}) => {
  return (
    <li className="list-item-container">
      <Link to={`/${vehicle.id}`}>
        {vehicle.title}
      </Link>
    </li>
  );
}

ListItem.propTypes = {
  vehicle: PropTypes.object.isRequired,
}

export default ListItem;