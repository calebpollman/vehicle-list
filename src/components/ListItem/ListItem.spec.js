import React from 'react';
import {shallow} from 'enzyme';
import ListItem from './ListItem';

describe('ListItem', () => {
  let wrapper;
  const vehicle = {
    "id": 0,
    "year" : "2013",
    "make" : "Kia",
    "model" : "Optima",
    "mileage" : 24235,
    "drivetrain" : "FWD",
    "bodytype" : "sedan",
    "image_url" : "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
    "created_at" : "2016-10-14T20:13:22.586Z"
  }

  beforeEach(() => {
    wrapper = shallow(<ListItem vehicle={vehicle} />)
  });

  it('should render a wrapper <li />', () => {
    expect(wrapper.find('li').length).toEqual(1);
  });
})