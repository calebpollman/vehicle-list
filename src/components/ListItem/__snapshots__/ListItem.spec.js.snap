// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    vehicle={
        Object {
            "bodytype": "sedan",
            "created_at": "2016-10-14T20:13:22.586Z",
            "drivetrain": "FWD",
            "id": 0,
            "image_url": "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
            "make": "Kia",
            "mileage": 24235,
            "model": "Optima",
            "year": "2013",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <li
        className="list-item-container"
>
        <p
                className="list-item-title"
        />
        <p
                className="list-item-text"
        >
                Mileage: 
                24235
        </p>
</li>,
      "className": "link-container",
      "replace": false,
      "to": "/0",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="list-item-title"
/>,
          <p
            className="list-item-text"
>
            Mileage: 
            24235
</p>,
        ],
        "className": "list-item-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "list-item-title",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Mileage: ",
              24235,
            ],
            "className": "list-item-text",
          },
          "ref": null,
          "rendered": Array [
            "Mileage: ",
            24235,
          ],
          "type": "p",
        },
      ],
      "type": "li",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <li
          className="list-item-container"
>
          <p
                    className="list-item-title"
          />
          <p
                    className="list-item-text"
          >
                    Mileage: 
                    24235
          </p>
</li>,
        "className": "link-container",
        "replace": false,
        "to": "/0",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="list-item-title"
/>,
            <p
              className="list-item-text"
>
              Mileage: 
              24235
</p>,
          ],
          "className": "list-item-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "list-item-title",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Mileage: ",
                24235,
              ],
              "className": "list-item-text",
            },
            "ref": null,
            "rendered": Array [
              "Mileage: ",
              24235,
            ],
            "type": "p",
          },
        ],
        "type": "li",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
